<launch>
    <!-- Define arguments -->
    <arg name="server_output" default="screen" doc="Output type for service server node (screen/log)"/>
    <arg name="client_output" default="screen" doc="Output type for service client node (screen/log)"/>
    <arg name="agent_name" default="turtle_1"/>
    <arg name="goal_x" default="8.0"/>
    <arg name="goal_y" default="8.0"/>
    <arg name="agent_number" default="1" doc="Number of agents to spawn"/>

    <!-- Launch turtlesim node -->
<include file="$(find service_server_client)/launch/service_server_client.launch">
      <arg name="server_output" value="log"/>
      <arg name="client_output" value="log"/>
      <arg name="agent_number" value="$(arg agent_number)"/>
</include>

    <!-- Launch service server node -->
    <node pkg="action_server_client" type="nav_action_server.py" name="nav_action_server" 
          output="$(arg server_output)">
    </node>
    <param name="goal_x" value="$(arg goal_x)"/>
    <param name="goal_y" value="$(arg goal_y)"/>
    <param name="agent_name" value="$(arg agent_name)"/>
 <group ns="$(arg agent_name)">  
    <!-- Launch service client node -->
    <node pkg="action_server_client" type="nav_action_client.py" name="nav_action_client" 
          output="$(arg client_output)">

    </node>
</group>
</launch>